### ISSUE TITLE
cy.type does not fire `beforeInput` event / does not work with slate.js

### ISSUE THREAD

**Issue by nekolnyt** [cypress-io/cypress#7088]
### Current behavior:
`cy.type` command does not fire `beforeInput` event.

The documentation https://docs.cypress.io/api/commands/type.html#Events says:
> `beforeinput` is not fired even though it is in the spec because no browser has adopted it.

This statement is not true anymore and not firing the event makes it impossible to use Cypress for testing editor implemented with Slate.js.

### Desired behavior:
`cy.type` does fire `beforeInput` event.

### Test code to reproduce

https://github.com/nekolnyt/cypress-test-tiny-type-in-slate/blob/master/cypress/integration/spec.js
### Versions
Cypress 4.4.0
OS: macOS Catalina 10.15.1
Browser: Chrome 81

###  Related issues
https://github.com/ianstormtaylor/slate/issues/3476
https://github.com/cypress-io/cypress/issues/311#issuecomment-610558564

**Comment by reydelo** [comment]
Hopefully the Cypress team will update their input commands to dispatch the beforeinput event, but until they do I’ve created a couple of simple custom commands which will trigger Slate’s input event listeners and make it respond https://github.com/ianstormtaylor/slate/issues/3476#issuecomment-617594068

**Comment by sainthkh** [comment]
After some research, I learned that simple `cy.get('[data-slate-editor="true"]').type('test')` is impossible. Because Cypress forcefully inserts text to content editable div. We need a new option like `noAutomaticContentEditableUpdate`. But the problem is that the name is too long. 

One more problem is the `inputType` field of `InputEvent`. We need to decide what is the good default value.

**Comment by sainthkh** [comment]
After reading [the list of `inputType`s](https://rawgit.com/w3c/input-events/v1/index.html#interface-InputEvent-Attributes) and checking which key combo triggers which, I decided to fix this issue after #5694.

Because in current Cypress 5+, simulating some key combos like `ctrl+backspace` is impossible. 

For example, the test below fails.

```js
  cy.get('#x').type('1234{ctrl}{backspace}')
  cy.get('#x').should('have.value', '')
```

But as we all know, ctrl+backspace deletes a word. So, it's a bug. 

After fixing this problem, I'll revisit this issue. Sorry for the wait.
